spec/queries/count_classifications_spec.rb

```diff
@@ -0,0 +1,93 @@
+    end
+  end
+
+  describe 'select_clause' do
+    it 'buckets counts by year by default' do
+      counts = count_classifications.call(params)
+      expected_select_query = "SELECT time_bucket('1 year', day) AS period, SUM(classification_count)::integer AS count FROM \"daily_classification_count\" GROUP BY period ORDER BY period"
+      expect(counts.to_sql).to eq(expected_select_query)
+    end
+
+    it 'buckets counts by given period' do
+      params[:period] = 'week'
+      counts = count_classifications.call(params)
+      expected_select_query = "SELECT time_bucket('1 week', day) AS period, SUM(classification_count)::integer AS count FROM \"daily_classification_count\" GROUP BY period ORDER BY period"
+      expect(counts.to_sql).to eq(expected_select_query)
+    end
+  end
+
+  describe '#call' do
+    let!(:classification_event) { create(:classification_event) }
+    let!(:diff_workflow_event) { create(:event_with_diff_workflow) }
+    let!(:diff_project_event) { create(:event_with_diff_project) }
+    let!(:diff_time_event) { create(:event_created_yesterday) }
+
+    it_behaves_like 'is filterable'
+
+    it 'returns counts of all events when no params given' do
+      counts = count_classifications.call(params)
+      # because default is bucket by year and all data created in the same year, we expect counts to look something like
+      # [<ClassificationCounts::DailyClassificationCount period: 01-01-2023, count: 4>]
+      current_year = Date.today.year
+      expect(counts.length).to eq(1)
+      expect(counts[0].count).to eq(4)
+      expect(counts[0].period).to eq("01-01-#{current_year}")
+    end
+
+    it 'returns counts bucketed by given period' do
+      params[:period] = 'day'
+      counts = count_classifications.call(params)
+      expect(counts.length).to eq(2)
+      expect(counts[0].count).to eq(1)
+      expect(counts[0].period).to eq((Date.today - 1).to_s)
+      expect(counts[1].count).to eq(3)
+      expect(counts[1].period).to eq(Date.today.to_s)
+    end
+
+    it 'returns counts of events with given workflow' do
+      workflow_id = diff_workflow_event.workflow_id
+      params[:workflow_id] = workflow_id.to_s
+      counts = count_classifications.call(params)
+      expect(counts.length).to eq(1)
+      expect(counts[0].count).to eq(1)
+    end
+
+    it 'returns counts of events with given project' do
+      project_id = diff_project_event.project_id
+      params[:project_id] = project_id.to_s
+      counts = count_classifications.call(params)
+      expect(counts.length).to eq(1)
+      expect(counts[0].count).to eq(1)
+    end
+
+    it 'returns counts of events within given date range' do
+      last_week = Date.today - 7
+      today = Date.today
+      params[:start_date] = last_week.to_s
+      params[:end_date] = today.to_s
+      counts = count_classifications.call(params)
+      expect(counts.length).to eq(1)
+      expect(counts[0].count).to eq(1)
```

Comment:
Because of how the `range_clause` is exclusive (using `<` and `>` instead of `>=` and `<=`), this "between start date and today" query only returns one record (what happened yesterday) and does not include what happened today (or on the start date, were there any). Requests for a range that starts and ends on dates (including "the last week" like this spec) but excludes the selected dates will be missing two days worth of expected counts. This might be counterintuitive when it comes to writing requests.